cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -221,6 +233,13 @@ private String appendPath(String base, String path) {
   }
 
   private void handleNotFound(PeerInfo peerInfo) throws IOException {
+    if (peerInfo.getTetheringStatus() != TetheringStatus.PENDING) {
```

Comment:
It seems like it should always go into rejected regardless of the existing state. Otherwise I can make a tethering request, the other side rejects it, then it automatically gets created again.

Reply:
fixed